### PR TITLE
[ONNX] reduce log spam when exporting dropout in training mode

### PR DESCRIPTION
**Pull request by garymm**
The default for `torch.onnx.export` is `TrainingMode.EVAL`:
https://github.com/pytorch/pytorch/blob/0d76299ff782224810b997d99f8af490cd96eede/torch/onnx/__init__.py#L63

That means that this warning is only printed when the caller overrides
that and explicitly specifies that they want training ops like Dropout.
We should assume the user knows what they're doing and not warn.

Also set `do_constant_folding=False` in the dropout related training tests. Without this, warnings are printed like:
```
UserWarning: It is recommended that constant folding be turned off ('do_constant_folding=False') when exporting the model in training-amenable mode
```

### CODE REVIEW

File: torch/onnx/symbolic_opset12.py [hunk 1]
```diff
@@ -51,15 +50,11 @@ def outer(g, input, other):
...
+    # if train is False, dropout is no-op
     if not train:
         return input
-    warnings.warn(
-        "Dropout is a training op and should not be exported in inference mode. "
-        "For inference, make sure to call eval() on the model and to export it with param training=False."
-    )
     p = g.op("Constant", value_t=torch.tensor(p))
-    t = g.op("Constant", value_t=torch.tensor(True))
+    t = g.op("Constant", value_t=torch.tensor(train))
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset12.py:57]: Maybe set dtype?

Comment (garymm) [reply torch/onnx/symbolic_opset12.py]: Done.

**Review by justinchuby — APPROVED** [review]
LGTM🚀

**Comment by justinchuby** [comment]
@BowenBao for approval

**Review by BowenBao — APPROVED** [review]
